src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/PremisesService.kt

```diff
@@ -91,4 +95,74 @@ class PremisesService(
 
       return success(lostBedsEntity)
     }
+
+  fun createNewPremises(requestBody: NewPremises) = validated<PremisesEntity> {
+
+    // TODO: dummy data for now - remove asap
+    val apAreaEntity = ApAreaEntity(
+      id = UUID.randomUUID(),
+      name = "arbitrary_ap_area",
+      identifier = "arbitrary_identifier",
+      probationRegions = mutableListOf()
+    )
+
+    val probationRegion = ProbationRegionEntity(
+      id = UUID.fromString("afee0696-8df3-4d9f-9d0c-268f17772e2c"), // Wales in db
+      name = "arbitrary_probation_region",
+      apArea = apAreaEntity,
+      premises = mutableListOf()
+    )
+    // end of dummy data
+
+    if (requestBody.address.isEmpty()) {
+      return generalError("Please provide an address")
+    }
+
+    if (requestBody.postcode.isEmpty()) {
+      return generalError("Please provide a postcode")
+    }
+
+    if (requestBody.service.isEmpty()) {
+      return generalError("Please provide a service type")
+    }
+
+    try {
+      UUID.fromString(requestBody.localAuthorityId)
```

Comment:
If you specify the type of `localAuthortiyId` in api.yml as:

```
type: string
format: uuid
```

OpenAPI generate will type the property as a UUID in the model it generates.  Then you should get this behaviour for free - e.g. here I just added a `aUUID` property to the create booking request

![image](https://user-images.githubusercontent.com/110207532/196403949-7dee4c5d-de61-447a-bc27-4f812b71cf3d.png)